app/src/main/java/edu/eci/ieti/myapplication/AddActivity.java

```diff
@@ -0,0 +1,117 @@
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class AddActivity extends AppCompatActivity {
+    private AppBarConfiguration mAppBarConfiguration;
+
+    private EditText editTextName;
+    private EditText editTextDepartment;
+    private EditText editTextCity;
+    private EditText editTextDirection;
+    private EditText editTextDescription;
+    private EditText editTextCapacity;
+    private EditText editTextHabitations;
+    private EditText editTextBathrooms;
+    private PlaceService placeService;
+
+    private final ExecutorService executorService = Executors.newFixedThreadPool(1);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
+                || super.onSupportNavigateUp();
+    }
+
+    public void onSubmitAdd(View view){
+        loadComponents();
+        String name = editTextName.getText().toString();
+        String department = editTextDepartment.getText().toString();
+        String city = editTextCity.getText().toString();
+        String direction = editTextDirection.getText().toString();
+        String description = editTextDescription.getText().toString();
+        String urlImg = "https://i.blogs.es/8e8f64/lo-de-que-comprar-una-casa-es-la-mejor-inversion-hay-generaciones-que-ya-no-lo-ven-ni-de-lejos---1/450_1000.jpg";
+        int capacity = Integer.parseInt(editTextCapacity.getText().toString());
+        int habitations = Integer.parseInt(editTextHabitations.getText().toString());
+        int bathrooms = Integer.parseInt(editTextBathrooms.getText().toString());
+
+        if (name.isEmpty()) {
+            editTextName.setError("El parametro de nombre no puede estar vacio");
+        } else {
+            executorService.execute(() -> {
+                try {
+                    Response<ResponseBody> response =
+                            placeService.addPlace(new Place(name,department,city,direction,description,urlImg,capacity,habitations,bathrooms),"bivianasanchez99@gmail.com").execute();
+                    runOnUiThread(() -> {
+                        if (response.isSuccessful()) {
+                            System.out.println("suuu");
+                        } else {
+                            editTextName.setError("Error de Conexion con BackEnd");
+                        }
+                    });
+                } catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
Configure a logger to properly report application errors. In addition, exceptions may be reported to the user if the programmer is not attempting to handle them.